webapp/loggedin/surveyViews/taxonomyEdit/components/taxonomyEdit.js

```diff
@@ -1,89 +1,156 @@
+      <div>{i18n.t('common.code')}</div>
+      <div>{i18n.t('taxonomy.edit.family')}</div>
+      <div>{i18n.t('taxonomy.edit.genus')}</div>
+      <div>{i18n.t('taxonomy.edit.scientificName')}</div>
+      {
+        vernacularLanguageCodes.map(lang =>
+          <div key={`vernacular_name_header_${Taxonomy.getUuid(taxonomy)}_${lang}`}>
+            {R.propOr(lang, lang)(languages)}
+          </div>
+        )
+      }
+    </>
+  )
+}
+const TaxonRow = ({ row, idx, offset, vernacularLanguageCodes }) => (
+  <>
+    <div>{idx + offset + 1}</div>
+    <div>{Taxon.getCode(row)}</div>
+    <div>{Taxon.getFamily(row)}</div>
+    <div>{Taxon.getGenus(row)}</div>
+    <div>{Taxon.getScientificName(row)}</div>
+    {
+      vernacularLanguageCodes.map(lang =>
+        <div key={`vernacular_name_${Taxon.getUuid(row)}_${lang}`}>
+          {Taxon.getVernacularName(lang)(row)}
+        </div>
+      )
     }
-  }, [Taxonomy.getUuid(taxonomy)])
+  </>
+)
 
+const TaxonEditHeader = props => {
+  const { surveyId, taxonomy, taxa, readOnly, putTaxonomyProp, uploadTaxonomyFile } = props
+  const i18n = useI18n()
   const validation = Validation.getValidation(taxonomy)
 
+  return (
+    <div className="taxonomy-edit__header">
+
+      <FormItem label={i18n.t('taxonomy.edit.taxonomyName')}>
+        <div>
+          <Input value={Taxonomy.getName(taxonomy)}
+                 validation={Validation.getFieldValidation('name')(validation)}
+                 onChange={value => putTaxonomyProp(taxonomy, 'name', StringUtils.normalizeName(value))}
+                 readOnly={readOnly} />
+        </div>
+      </FormItem>
+
+      <div className="button-bar">
+        {
+          !readOnly &&
+          <UploadButton label={i18n.t('common.csvImport')}
+                        disabled={Taxonomy.isPublished(taxonomy)}
+                        accept=".csv"
+                        onChange={async ([file]) => { await uploadTaxonomyFile(taxonomy, file) } }/>
+
+        }
+        <DownloadButton href={`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}/export?draft=true`}
+                        disabled={R.isEmpty(taxa)}
+                        label={i18n.t('common.csvExport')} />
+      </div>
+    </div>
+  )
+}
+
+const TaxonomyEdit = props => {
+
+  const {
+    surveyId, taxonomy, taxa,
+    setTaxonomyForEdit,
+    readOnly,
+    putTaxonomyProp,
+    uploadTaxonomyFile,
+    initList,
+    activeJob,
+  } = props
+
   const i18n = useI18n()
 
-  return (
-    <div className="taxonomy-edit">
+  const taxonomyUuid = Taxonomy.getUuid(taxonomy)
 
-      <div className="taxonomy-edit__header">
+  const vernacularLanguageCodes = R.reduce(
+    (acc, taxon) => R.concat(acc, R.difference(R.keys(Taxon.getVernacularNames(taxon)), acc)),
+    [],
+    taxa
+  )
 
-        <ErrorBadge validation={validation}/>
+  useOnUpdate(() => {
+    if (!activeJob) {
+      initList(TaxonomyEditState.keys.taxa, moduleApiUri, moduleApiCountUri)
```

Comment:
this should be done in the actions, on job complete.
In fact, when I updated a list, the tables is blank